cpp/dolfinx/mesh/Mesh.h

```diff
@@ -101,6 +101,16 @@ class Mesh
   /// @return The communicator on which the mesh is distributed
   MPI_Comm comm() const;
 
+  /// Create a submesh consisting of a subset of entities in the
+  /// mesh (this)
+  /// @param[in] dim Entity dimension
+  /// @param[in] entities Vector of entities in the mesh
+  /// @return The submesh, a map from the (local) submesh vertices
+  /// to the (local) mesh vertices, and a map from the (local)
+  /// submesh geometry dofs to the (local) mesh geometry dofs
+  std::tuple<Mesh, std::vector<std::int32_t>, std::vector<std::int32_t>>
+  create_submesh(int dim, const xtl::span<const std::int32_t>& entities);
```

Comment:
Does this need to be a class method, or could it be a free function? Free functions are generally preferable. Methods are appropriate when access to private data is required.